live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/config/jackson/UpdateNodeEventSerializer.java

```diff
@@ -96,6 +107,35 @@ public void serialize(UpdateNodeEvent event, JsonGenerator jgen, SerializerProvi
         }
     }
 
+    private NodeProperty<?> filterNonEmptyValueAndMap(NodeProperty<?> property)
```

Comment:
Wouldn't it be neater if you returned `Optional`? That way you could get rid of those `null`s getting returned inside conditionals.

Reply:
Good hint! Fixed

Reply:
I went event further and finally even got rid of the `Optional`